web/src/search_suggestion.js

```diff
@@ -473,10 +514,24 @@ function get_special_filter_suggestions(last, operators, suggestions) {
     // Negating suggestions on is_search_operand_negated is required for
     // suggesting negated operators.
     if (last.negated || is_search_operand_negated) {
+        // We expect all suggestions to only have one part at this point.
+        // TODO: Ideally we'd have types to differentiate suggestions
+        // that are definitely single-part and ones that are potentially
+        // multi-part, and have clear separation in the code of where
+        // each is used.
```

Comment:
I'm still a little confused about how the suggestion data works, and haven't looked in depth into how this data evolves over the lifecycle of creating the search suggestions over time. 

This PR essentially turns every suggestion into something that could contain multiple parts (such as a user pill), and I've tested it manually for several cases that all seem to work fine.

It does seem like some places expect suggestions to only have one part (previously that assumption would be "some places expect suggestions to have a search string with only one clause in it") and this is one of those places, and you can see that assumption in things like `"-" + suggestion.search_string,`.

I can look further into the lifecycle of how search suggestions are built if that's helpful, or someone who's more familiar with it can chime in as part of review.